Source/RxStarscream.swift

```diff
@@ -8,141 +8,142 @@ import RxCocoa
+        subject.on(.next(WebSocketEvent.disconnected(error)))
+        forwardDelegate?.websocketDidDisconnect(socket: socket, error: error)
+    }
+
+    public func websocketDidReceiveMessage(socket: WebSocket, text: String) {
+        subject.on(.next(WebSocketEvent.message(text)))
+        forwardDelegate?.websocketDidReceiveMessage(socket: socket, text: text)
+    }
+
+    public func websocketDidReceiveData(socket: WebSocket, data: Data) {
+        subject.on(.next(WebSocketEvent.data(data)))
+        forwardDelegate?.websocketDidReceiveData(socket: socket, data: data)
+    }
+
+    public func websocketDidReceivePong(socket: WebSocket, data: Data?) {
+        subject.on(.next(WebSocketEvent.pong))
+        forwardPongDelegate?.websocketDidReceivePong(socket: socket, data: data)
+    }
+
+    deinit {
+        subject.on(.completed)
+    }
 }
 
 extension Reactive where Base: WebSocket {
-  
-  public var response: Observable<WebSocketEvent> {
-    return RxWebSocketDelegateProxy.proxyForObject(base).subject
-  }
-  
-  public var text: Observable<String> {
-    return self.response.filter { response in
-      switch response {
-      case .message(_):
-        return true
-      default:
-        return false
-      }
-    }.map { response in
-      switch response {
-      case .message(let message):
-        return message
-      default:
-        return String()
-      }
+
+    public var response: Observable<WebSocketEvent> {
+        return RxWebSocketDelegateProxy.proxyForObject(base).subject
     }
-  }
-  
-  public var connected: Observable<Bool> {
-    return response.filter { response in
-      switch response {
-      case .connected:
-        return true
-      case .disconnected(_):
-        return true
-      default:
-        return false
-      }
-      }.map { response in
-        switch response {
-        case .connected:
-          return true
-        default:
-          return false
+
+    public var text: Observable<String> {
+        return self.response.filter { response in
+            switch response {
+            case .message(_):
+                return true
+            default:
+                return false
+            }
+        }.map { response in
+            switch response {
+            case .message(let message):
+                return message
+            default:
+                return String()
+            }
         }
     }
-  }
-
-  public func write(data: Data) -> Observable<Void> {
-    return Observable.create { sub in
-      self.base.write(data: data) {
-        sub.onNext(())
-        sub.onCompleted()
-      }
-      return Disposables.create()
+
+    public var connected: Observable<Bool> {
+        return response.filter { response in
+            switch response {
+            case .connected:
+                return true
+            case .disconnected(_):
+                return true
+            default:
+                return false
+            }
+        }.map { response in
```

Comment:
Looking here I don't see the point of the filter before this? Couldn't we just have the map ?